langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal

```diff
@@ -2440,6 +2494,23 @@ function testCloneWithTypeToUnion() {
     assertFalse(tupleValue is decimal);
     assertFalse(tupleValue is [string, int]);
     assertFalse(tupleValue is error);
+
+    json extCoding = {
+        "valueCoding": {
+            "system": "http://loinc.org",
+            "code": "LA29518-0",
+            "display": "he/him/his/himself"
+        },
+        "url": "http://open.epic.com/FHIR/StructureDefinition/extension/calculated-pronouns-to-use-for-text"
+    };
+
+    Extension ext = checkpanic extCoding.cloneWithType();
+    assertEquality(ext, <Extension>{
+        "url":
+    "http://open.epic.com/FHIR/StructureDefinition/extension/calculated-pronouns-to-use-for-text",
+        "valueCoding": {"system": "http://loinc.org", "code": "LA29518-0", "display": "he/him/his/himself"}
+    });
+    assertEquality((typeof ext).toString(), "typedesc CodingExtension");
```

Comment:
Shall we use the `is` check in addition to the `typeof`? because the `typeof` return values vary as per immutability.

Reply:
Since we need to add multiple is checks, will keep in this way as discussed offline.